app/v2/controllers/actions/AcceptHeaderAction.scala

```diff
@@ -55,7 +56,13 @@ class AcceptHeaderActionImpl[R[_] <: Request[_]] @Inject() (implicit val executi
     }
 
   private def checkAcceptHeader[A](acceptHeaderValue: String, request: R[A]): Either[Result, R[A]] =
-    if (acceptedHeaders.contains(acceptHeaderValue)) Right(request)
-    else Left(NotAcceptable(Json.toJson(PresentationError.notAcceptableError(s"Accept header $acceptHeaderValue is not supported!"))))
+    if (
+      (!isOnlyJson && acceptedHeaders
+        .contains(acceptHeaderValue)) || (isOnlyJson && VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON.contains(acceptHeaderValue))
+    ) {
```

Comment:
This feels a bit messy to me -- once an if statement goes beyond two conditions it's worth starting to think about how this might be done better.

The goal of the `isOnlyJson` boolean is to limit the accepted headers to the Json header. So, instead of making this change here and doing this conditional every time, I suggest reverting this method back to what it was before, and changing the `acceptedHeaders` lazy val, like so:

```scala
private lazy val acceptedHeaders = 
    if (isOnlyJson) Seq(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON) 
    else Seq(
        VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON,
        VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML,
        VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML_HYPHEN
    )
```

That way we initialise the `acceptedHeaders` once, store the list, and the `checkAcceptHeader` function can continue to check this seq.